plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2488,9 +2495,13 @@ void producer_plugin_impl::schedule_production_loop() {
       chain::controller& chain = chain_plug->chain();
       fc_dlog(_log, "Speculative Block Created; Scheduling Speculative/Production Change");
       EOS_ASSERT(chain.is_building_block(), missing_pending_block_state, "speculating without pending_block_state");
-      auto wake_time = block_timing_util::calculate_producer_wake_up_time(_produce_block_cpu_effort, chain.pending_block_num(), chain.pending_block_timestamp(),
+      auto wake_time = block_timing_util::calculate_producer_wake_up_time(fc::microseconds{config::block_interval_us}, chain.pending_block_num(), chain.pending_block_timestamp(),
                                                                           _producers, chain.head_block_state()->active_schedule.producers,
                                                                           _producer_watermarks);
+      if (wake_time && fc::time_point::now() > *wake_time) {
+         // if wake time has already passed then use the block deadline instead
+         wake_time = _pending_block_deadline;
```

Comment:
should we check that `_pending_block_deadline` > `fc::time_point::now()`?

Reply:
I mean, if `_pending_block_deadline` < `fc::time_point::now()`, we would still be setting a `wake_time ` that is in the past.

Reply:
Right, setting it in the past will mean it immediately will wake up and start a new block. If the pending deadline has passed then that is what we want I believe.

Reply:
OK, perfect, thanks!